actors.md

```diff
@@ -688,68 +675,36 @@ func SubmitPost(proofs PoStProof, faults []FaultSet, recovered Bitfield, done Bi
 		TransferFunds(msg.From, msg.Value-feesRequired)
 	}
 
-	if !CheckPostProof(miner.SectorSize, proof, faults) {
+    seed := GetRandFromBlock(self.ProvingPeriodStart + ProvingPeriodDuration(self.SectorSize) - POST_CHALLENGE_TIME)
+    faultSet := self.currentFaultSet
+
+	if !VerifyPoSt(self.SectorSize, self.provingSet, seed, proof, faultSet) {
 		Fatal("proof invalid")
 	}
 
-	// combine all the fault set bitfields, and subtract out the recovered
-	// ones to get the set of sectors permanently lost
-	permLostSet = allFaults.Subtract(recovered)
+    // The next fault set becomes the current one
+    self.currentFaultSet = self.nextFaultSet
+    self.nextFaultSet = EmptySectorSet()
 
-	// burn funds for fees and collateral penalization
-	self.BurnFunds(CollateralForSize(self.SectorSize*permLostSet.Size()) + feesRequired)
+    // TODO: penalize for faults
```

Comment:
too dooo

Reply:
yeah, this will stay for now and a new issue for handling post faults will be created